nuxeo-features/nuxeo-platform-oauth/src/main/java/org/nuxeo/ecm/platform/oauth2/clients/OAuth2ClientServiceImpl.java

```diff
@@ -66,6 +72,44 @@ public OAuth2Client getClient(String clientId) {
         return queryClients().stream().map(OAuth2Client::fromDocumentModel).collect(Collectors.toList());
     }
 
+    @Override
+    public OAuth2Client create(OAuth2Client oAuth2Client) {
+        Objects.requireNonNull(oAuth2Client, "oAuth2Client is required");
+        if (getClientModel(oAuth2Client.getId()) != null) {
```

Comment:
I wonder if this code shouldn't be in the `OAuth2Client#validate` method, since you also want to perform this check for an update.

Reply:
the control in Create/Update it's not the same. in Create is to be sur that the client is unique and in the update that the client that we want to update is exist. i put in the validate method the commons code

Reply:
Yes, when you update, you first check that the client corresponding to the `"clientId"` path parameter exists.
But then, you also want to check that the updated client id (`OAuth2Client#getId` on the `client` parameter) doesn't match an existing client.